### PR TITLE
Csa and csd reconstruction workflow rebased

### PR DESCRIPTION
**Pull request by Garyfallidis**
This is a rebase of https://github.com/nipy/dipy/pull/1155
with some additional testing and using the new hdf5 format (.pam5) for saving PeaksAndMetrics.
@matthieudumont, @arokem and all give it a try.

### CODE REVIEW

**Comment by Garyfallidis** [comment]
Hi, this looks ready can you try it with your datasets? And tell me if it works for you?

**Comment by matthieudumont** [comment]
(sorry for long response time)
Looks good and works for me, merge it!

**Comment by Garyfallidis** [comment]
Perfeto! Thank you @matthieudumont. @arokem can you try it too and merge?

File: setup.py [hunk 1]
```diff
@@ -218,6 +218,8 @@ def main(**extra_args):
           scripts      = [pjoin('bin', 'dipy_reconst_dti'),
                           pjoin('bin', 'dipy_reconst_dti_restore'),
                           pjoin('bin', 'dipy_mask'),
+                          pjoin('bin', 'dipy_reconst_csa'),
+                          pjoin('bin', 'dipy_reconst_csd'),
```
Comment (arokem) [inline setup.py]: This is still rather ad-hoc. We *need* to automate this.

Comment (arokem) [inline setup.py]: https://github.com/nipy/dipy/issues/1253

**Pull request merged**